ewf.go

```diff
@@ -116,24 +116,24 @@ func (rp *RetryPolicy) UnmarshalJSON(data []byte) error {
 }
 
 // BeforeWorkflowHook is a function run before a workflow starts.
-type BeforeWorkflowHook func(ctx context.Context, w *Workflow)
+type BeforeWorkflowHook func(ctx context.Context, w Workflow)
 
 // AfterWorkflowHook is a function run after a workflow finishes.
-type AfterWorkflowHook func(ctx context.Context, w *Workflow, err error)
+type AfterWorkflowHook func(ctx context.Context, w Workflow, err error)
 
 // BeforeStepHook is a function run before a step starts.
-type BeforeStepHook func(ctx context.Context, w *Workflow, step *Step)
+type BeforeStepHook func(ctx context.Context, w Workflow, step *Step)
```

Comment:
it should be still passed as a pointer in case the step wanted to mutate it. there should be no concurrent access here to the workflow since steps and hooks run in the same goroutine